text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -0,0 +1,114 @@
++ Feature name: dust-protection
++ Start date: 2020-12-07
++ RFC PR: [iotaledger/protocol-rfcs#0032](https://github.com/iotaledger/protocol-rfcs/pull/0032)
+
+
+# Summary
+
+This RFC defines a new protocol rule regarding processing outputs that transfer a small amount of Iotas, known as dust outputs. The rule states that each dust output should be backed up by a deposit of the receiving address. Thus making it expensive to proliferate dust. Since a receiver must make a deposit, the protocol makes receiving dust an opt-in feature.
+
+# Motivation
+
+The total supply of IOTA is approximately 2.78 peta IOTA. Considering that a single typical `SigLockedSingleOutput` takes 42 bytes (just the plain output bytes without the encompassing transaction), it is easy to see how the size of the UTXO set may pose a potential problem. An attacker splitting 1 GI to single iota outputs can blow up the set size so that it won't fit into 4GB of RAM. This is far from being a prohibitively expensive attack.
+
+Thus, in order to protect nodes from such memory attacks we should make accumulating small dust outputs expensive. Other DLTs have fees that provide protection, but IOTA is feeless. So instead of fees we can have deposits. 
+
+When an address wants to receive micro transactions it must have a special output locked as a deposit. This deposit can't be consumed by any transaction as long as the dust outputs exist on the address. This also prevents attackers from sending dust to addresses that didn't opt into allowing dust outputs.
+
+Additional benefit of this rule is that it makes a mass of privacy violating [forced address reuse attacks](https://en.bitcoin.it/wiki/Privacy#Forced_address_reuse) more expensive to carry out.
+
+
+# Detailed design
+
+### Definitions
+
+*Dust*: a UTXO that spends an amount smaller than 1 MI.
+
+*SigLockedDustAllowanceOutput* - A new output type for deposits that allow an address to recieve dust outputs. It can be consumed as an input like a regular SigLockedOutput.
```

Comment:
Who can generate this output for the address? An address that does not want to support dust could be forced into it.

Reply:
Yes, a 3rd party can create a deposit on your address. But it this really an issue? You have the option to just ignore any dust and SigLockedDustAllowanceOutput and only consider your expected transactions. This would then effectively be equivalent to forbidding foreign SigLockedDustAllowanceOutput. Or, you even have the option to siphon the dust and make (small) profit. 

Forbidding foreign SigLockedDustAllowanceOutput would be possible but also complicate the validation and introduce some strange criteria here. I think, we talked about this, but keeping everything as simple as possible seemed more desirable especially since it is not a big drawback.

However, the least we should do is mention this and our motivation.